src/pages/active.tsx

```diff
@@ -1,87 +1,69 @@
-import { FC, useState, useEffect } from 'react';
-import Head from '@/components/head';
-import Layout from '@/components/Layout';
-import Card from '@/components/tasks/card';
-import useFetch from '@/hooks/useFetch';
-import classNames from '@/styles/tasks.module.scss';
-import task from '@/interfaces/task.type';
+import { FC, useState, useEffect } from "react";
+import Head from "@/components/head";
+import Layout from "@/components/Layout";
+import Card from "@/components/tasks/card";
+import useFetch from "@/hooks/useFetch";
+import classNames from "@/styles/tasks.module.scss";
+import task from "@/interfaces/task.type";
 import {
-  BLOCKED, IN_PROGRESS, SMOKE_TESTING,
-} from '@/components/constants/task-status';
-import updateTasksStatus from '@/helperFunctions/updateTasksStatus';
-import beautifyTaskStatus from '@/helperFunctions/beautifyTaskStatus';
+  BLOCKED,
+  IN_PROGRESS,
+  SMOKE_TESTING,
+} from "@/components/constants/task-status";
+import updateTasksStatus from "@/helperFunctions/updateTasksStatus";
+import beautifyTaskStatus from "@/helperFunctions/beautifyTaskStatus";
 
-const TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
+const MY_TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
 
 const renderCardList = (tasks: task[]) => {
   const beautifiedTasks = beautifyTaskStatus(tasks);
-  return beautifiedTasks.map(
-    (item: task) => (
-      <Card
-        content={item}
-        key={item.id}
-        shouldEdit={false}
-        onContentChange={undefined}
-      />
-    ),
-  );
+  return beautifiedTasks.map((item: task) => (
+    <Card
+      content={item}
+      key={item.id}
+      shouldEdit={false}
+      onContentChange={undefined}
+    />
+  ));
 };
 
 const Active: FC = () => {
   const [tasks, setTasks] = useState<task[]>([]);
   const [activeTasks, setActiveTasks] = useState<any>(null);
-  const statusActiveList = [
-    BLOCKED,
-    IN_PROGRESS,
-    SMOKE_TESTING,
-  ];
-  const {
-    response,
-    error,
-    isLoading,
-  } = useFetch(TASKS_URL);
+  const statusActiveList = [BLOCKED, IN_PROGRESS, SMOKE_TESTING];
+  const { response, error, isLoading } = useFetch(MY_TASKS_URL);
 
   useEffect(() => {
-    if ('tasks' in response) {
+    if ("tasks" in response) {
       setTasks(updateTasksStatus(response.tasks));
-      const active = tasks.filter(
-        (item: task) => (statusActiveList.includes(item.status)),
+      const active = tasks.filter((item: task) =>
+        statusActiveList.includes(item.status)
       );
       setActiveTasks(active);
     }
-  // eslint-disable-next-line react-hooks/exhaustive-deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why is this rule disabled?

Reply:
Actually that's the way to set the rule, you can get more details from this resource:  https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528 
and the reason which I can correlate with this current code is,  if you need to fetch data from a server, you usually pass to useEffect an empty array as a second argument (because you need to run the callback only once when your component did mount) - and it's an official advice from React documentation. And then "exhaustive-deps" often starts to annoy you with warnings that you "forgot" dependencies (but you're sure you didn't). And adding dependencies you don't need is an antipattern, because it may lead to unnecessary rerenderings or even to an infinite cycle of rerenderings. 
So we have to disable the rule in such cases.